Source/Core/Core/HW/EXI/BBA/BuiltIn.cpp

```diff
@@ -729,5 +731,67 @@ BbaUdpSocket::BbaUdpSocket() = default;
 
 sf::Socket::Status BbaUdpSocket::Bind(u16 port, u32 net_ip)
 {
-  return this->bind(port, sf::IpAddress(ntohl(net_ip)));
+  if (port != Common::SSDP_PORT)
+    return this->bind(port, sf::IpAddress(ntohl(net_ip)));
+
+  // Handle SSDP multicast
+  create();
+  const int on = 1;
+  if (setsockopt(getHandle(), SOL_SOCKET, SO_REUSEADDR, reinterpret_cast<const char*>(&on),
+                 sizeof(on)) != 0)
+  {
+    ERROR_LOG_FMT(SP1, "setsockopt failed to reuse SSDP address");
+  }
+#ifdef SO_REUSEPORT
+  if (setsockopt(getHandle(), SOL_SOCKET, SO_REUSEPORT, reinterpret_cast<const char*>(&on),
+                 sizeof(on)) != 0)
+  {
+    ERROR_LOG_FMT(SP1, "setsockopt failed to reuse SSDP port");
+  }
+#endif
+  if (const char loop = 1;
+      setsockopt(getHandle(), IPPROTO_IP, IP_MULTICAST_LOOP, &loop, sizeof(loop)) != 0)
+  {
+    ERROR_LOG_FMT(SP1, "setsockopt failed to set SSDP loopback");
+  }
+
+  // sf::UdpSocket::bind will close the socket and get rid of its options
+  sockaddr_in addr;
+  addr.sin_addr.s_addr = INADDR_ANY;
+  addr.sin_family = AF_INET;
+  addr.sin_port = htons(Common::SSDP_PORT);
+  Common::ScopeGuard error_guard([this] { close(); });
+  if (::bind(getHandle(), reinterpret_cast<const sockaddr*>(&addr), sizeof(addr)) != 0)
+  {
+    addr.sin_addr.s_addr = net_ip;
+    if (::bind(getHandle(), reinterpret_cast<const sockaddr*>(&addr), sizeof(addr)) != 0)
+      return sf::Socket::Status::Error;
```

Comment:
Can we get what the actual error was and log it?

Reply:
Sure, I'll probably use https://github.com/dolphin-emu/dolphin/pull/11003. I'll perform some testing before hand to see if it's worth returning a `string` or a `const char*` directly.